hibernate-core/src/test/java/org/hibernate/orm/test/sorted/SortTest.java

```diff
@@ -0,0 +1,169 @@
+	}
+
+	@Test
+	public void testSortedSetDefinitionInHbmXml() {
+		final PersistentClass entityMapping = metadata().getEntityBinding( Search.class.getName() );
+
+		final Property sortedSetProperty = entityMapping.getProperty( "searchResults" );
+		final Collection sortedSetMapping = assertTyping( Collection.class, sortedSetProperty.getValue()  );
+		assertTrue( "SortedSet mapping not interpreted as sortable", sortedSetMapping.isSorted() );
+		assertNull( sortedSetMapping.getComparator() );
+
+		final Property sortedMapProperty = entityMapping.getProperty( "tokens" );
+		final Collection sortedMapMapping = assertTyping( Collection.class, sortedMapProperty.getValue()  );
+		assertTrue( "SortedMap mapping not interpreted as sortable", sortedMapMapping.isSorted() );
+		assertNull( sortedSetMapping.getComparator() );
+
+		final Property sortedSetCaseInsensitiveProperty = entityMapping.getProperty( "searchResultsCaseInsensitive" );
+		final Collection sortedSetCaseInsensitiveMapping = assertTyping( Collection.class, sortedSetCaseInsensitiveProperty.getValue() );
+		assertTrue( "SortedSet mapping not interpreted as sortable", sortedSetCaseInsensitiveMapping.isSorted() );
+		assertTyping( StringCaseInsensitiveComparator.class, sortedSetCaseInsensitiveMapping.getComparator() );
+
+		final Property sortedMapCaseInsensitiveProperty = entityMapping.getProperty( "tokensCaseInsensitive" );
+		final Collection sortedMapCaseInsensitiveMapping = assertTyping( Collection.class, sortedMapCaseInsensitiveProperty.getValue() );
+		assertTrue( "SortedMap mapping not interpreted as sortable", sortedMapCaseInsensitiveMapping.isSorted() );
+		assertTyping( StringCaseInsensitiveComparator.class, sortedMapCaseInsensitiveMapping.getComparator() );
+
+	}
+
+	@Test
+	@FailureExpected( reason = "failure due to Criteria regression issue" )
+	public void testOrderBy() {
+		Search search = new Search( "Hibernate" );
+		search.searchResults.add( "jboss.com" );
+		search.searchResults.add( "hibernate.org" );
+		search.searchResults.add( "HiA" );
+
+		inTransaction(
+			sess -> {
+				sess.persist( search );
+				sess.flush();
+
+				sess.clear();
+				CriteriaBuilder criteriaBuilder = sess.getCriteriaBuilder();
+				CriteriaQuery<Search> criteria = criteriaBuilder.createQuery( Search.class );
+				Root<Search> root = criteria.from( Search.class );
+				criteria.select( root );
+
+				Search s = sess.createQuery( criteria ).uniqueResult();
+				assertFalse( Hibernate.isInitialized( s.searchResults ) );
+
+				assertThat( s.searchResults, IsIterableContainingInOrder.contains(
+						"HiA",
+						"hibernate.org",
+						"jboss.com"
+				) );
+
+				sess.clear();
+				criteria = criteriaBuilder.createQuery( Search.class );
+				root = criteria.from( Search.class );
+				root.fetch( "searchResults", JoinType.LEFT );
+				criteria.select( root );
+				s = sess.createQuery( criteria ).uniqueResult();
+				assertTrue( Hibernate.isInitialized( s.searchResults ) );
```

Comment:
Actually the above statement is the only reason `@FailureExpected` is used. Seems some complicated reason because of the query cache for Criteria. 
Given it is a good testing case for Criteria, I retained it though it seems laziness has nothing to do with the testing intention for this method.